### PR TITLE
Sage Charts MVP

### PR DESCRIPTION
**Pull request by philschanely**
## Description

This PR adds the following in:

- [Recharts](https://recharts.org/en-US) library to `@kajabi/sage-react` packages (already in use in `kajabi-products`)
- Chart React component along with a an MVP Bar and Donut type
- Components in React and Rails (for documentation) for Chart Summary and Legend
- Modification to TooltipElement to allow a `--static` variation that allows for use Charts
- Tokens and vendor style overrides that support Charts.
- new `clamp()` Sass mixin

### Basic Bar Chart

Screen Shot 2021-05-14 at 12 09 43 PM

### Multiple Bars Chart

Screen Shot 2021-05-14 at 12 09 32 PM

### Basic Donut Chart

Screen Shot 2021-05-14 at 12 10 02 PM

### Default Empty State

Screen Shot 2021-05-14 at 12 09 51 PM

## Testing in `sage-lib`

See new Chart components documentation in main Sage site.
See Storybook > Charts for demos of the basic charts available.

## Testing in `kajabi-products`
1. (LOW) Adds new Chart components. No impact on existing components.
1. (LOW) Modification to React Tooltip to allow for static variation. No impact on existing uses. Verify the following behave as expected:
   - [ ] In product outline, hover over a post to see the small set of icons appear then hover over them to see Tooltips appear

### CODE REVIEW

**Review by kimjaej — APPROVED** [review]
Well done 🚀

**Review by dangkevin — APPROVED** [review]
LGTM! this looks really nice!

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM, though it is concerning that the rechart library is lacking in accessibility support. The axis labels and legends are fully readable being text content and/or sage components, but the chart data values themselves appear to be obfuscated within the SVG. Nothing actionable for the MVP, but I wonder it's possible to follow up with a workaround (assuming it's possible given the library).

File: packages/sage-assets/lib/stylesheets/components/_chart_summary.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  pointer-events: none;
+}
+
+.sage-chart-summary--centered {
+  grid-column: 1 / -1;
+  grid-row: 1 / -1;
+}
+
+.sage-chart-summary--donut {
+  max-width: 120px;
```
[the remark this answers is not in the document]
Comment (philschanely) [reply packages/sage-assets/lib/stylesheets/components/_chart_summary.scss:24]: Good point. Its definitely a MVP fixture we can change as we know more about what donuts need moving forward. Its fixed now to set inside the donut, and the donut is configurable... hmm... perhaps I'll look into using some custom props to pass in at the component level in a follow up. 🤔

**Comment by philschanely** [comment]
@monkeypox8 good call on the a11y. It was only working with Recharts since it is what we had in `products` at present. I agree this is an issue we should revisit with the cross functional team too!

**Pull request merged**